gtk/DetailsDialog.cc

```diff
@@ -1867,29 +1903,35 @@ void appendAnnounceInfo(tr_tracker_view const& tracker, time_t const now, Gtk::T
 
         if (tracker.lastAnnounceSucceeded)
         {
-            gstr << gtr_sprintf(
-                _("Got a list of %1$s%2$'d peers%3$s %4$s ago"),
-                success_markup_begin,
-                tracker.lastAnnouncePeerCount,
-                success_markup_end,
-                timebuf);
+            gstr << fmt::format(
+                // {markup_begin} and {markup_end} should surround the peer text
+                ngettext(
+                    "Got a list of {markup_begin}{peer_count} peer{markup_end} {time_span} ago",
+                    "Got a list of {markup_begin}{peer_count} peers{markup_end} {time_span} ago",
+                    tracker.lastAnnouncePeerCount),
+                fmt::arg("markup_begin", SuccessMarkupBegin),
+                fmt::arg("peer_count", tracker.lastAnnouncePeerCount),
+                fmt::arg("markup_end", SuccessMarkupEnd),
+                fmt::arg("time_span", timebuf));
         }
         else if (tracker.lastAnnounceTimedOut)
         {
-            gstr << gtr_sprintf(
-                _("Peer list request %1$stimed out%2$s %3$s ago; will retry"),
-                timeout_markup_begin,
-                timeout_markup_end,
-                timebuf);
+            gstr << fmt::format(
+                // {markup_begin} and {markup_end} should surround the time_span
+                _("Peer list request {markup_begin}timed out {time_span} ago{markup_end}; will retry"),
+                fmt::arg("markup_begin", TimeoutMarkupBegin),
+                fmt::arg("time_span", timebuf),
+                fmt::arg("markup_end", TimeoutMarkupEnd));
         }
         else
         {
-            gstr << gtr_sprintf(
-                _("Got an error %1$s\"%2$s\"%3$s %4$s ago"),
-                err_markup_begin,
-                tracker.lastAnnounceResult,
-                err_markup_end,
-                timebuf);
+            gstr << fmt::format(
+                // {markup_begin} and {markup_end} should surround the error
+                _("Got an error {markup_begin}{error}{markup_end} {time_span} ago"),
```

Comment:
Maybe still quote the error since it's in the middle of the phrase? Alternatively, mark it up with more than just color (e.g. `<em>`).

Reply:
4665ebc22aec09a40dbbcdf1e9d276c33db46d0f